src/main/java/io/mosip/mimoto/service/impl/PresentationServiceImpl.java

```diff
@@ -74,16 +73,17 @@ public String authorizePresentation(PresentationRequestDTO presentationRequestDT
                 .stream()
                 .findFirst()
                 .map(inputDescriptorDTO -> {
-                    boolean matchingProofTypes = inputDescriptorDTO.getFormat().getLdpVc().getProofTypes()
+                    boolean matchingProofTypes = inputDescriptorDTO.getFormat().get("ldpVc").get("proofTypes")
                             .stream()
                             .anyMatch(proofType -> vcCredentialResponse.getCredential().getProof().getType().equals(proofType));
                     if (matchingProofTypes) {
                         logger.info("Started the Construction of VP token");
                         try {
-                            String vpToken = constructVerifiablePresentationString(vcCredentialResponse.getCredential());
-                            String presentationSubmission = constructPresentationSubmission(vpToken);
-                            return String.format(injiVerifyRedirectUrl,
-                                    presentationRequestDTO.getRedirect_uri(),
+                            VerifiablePresentationDTO verifiablePresentationDTO = constructVerifiablePresentationString(vcCredentialResponse.getCredential());
+                            String presentationSubmission = constructPresentationSubmission(verifiablePresentationDTO, presentationDefinitionDTO, inputDescriptorDTO);
+                            String vpToken = objectMapper.writeValueAsString(verifiablePresentationDTO);
+                            return String.format(injiOvpRedirectURLPattern,
+                                    presentationRequestDTO.getRedirectUri(),
                                     Base64.getUrlEncoder().encodeToString(vpToken.getBytes(StandardCharsets.UTF_8)),
                                     URLEncoder.encode(presentationSubmission, StandardCharsets.UTF_8));
```

Comment:
Since presentation Submission is sent via URL as a redirect, we need to send it url encoded (percent-encoded).
https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.3
https://swagger.io/specification/

As per these spec, generally it should be url encoded, since url does not directly fit { , } these

Reply:
ok